c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -1053,12 +1066,13 @@ class DeviceCachingAllocator {
 
       std::string proc_info = reportProcessMemoryInfo(device);
 
-      if (record_history) {
+      if (record_history || enable_trace_tracker_) {
```

Comment:
Now the condition for calling `record_trace` is more than a single bool, we should refactor to move the check into the `record_trace` function. The other stuff that is done within these blocks (e.g. block->context_when_allocated = std::move(context)`) is cheap enough to just run universally.